pkg/iosource/iosource.go

```diff
@@ -0,0 +1,100 @@
+package iosource
+
+import (
+	"errors"
+	"io"
+	"net/url"
+	"sync"
+)
+
+const FileScheme = "file"
+
+var (
+	DefaultRegistry = &Registry{}
+)
+
+func init() {
+	resetDefaultRegistry()
+}
+
+func resetDefaultRegistry() {
+	DefaultRegistry = &Registry{}
+	DefaultRegistry.Add(FileScheme, DefaultFileSource)
+}
```

Comment:
```suggestion
var DefaultRegistry = &Registry{
	schemes: map[string]Source{"file": DefaultFileSource},
}
```
And I think this file will read better if you put this right before the functions that use it (namely, immediately before Register).

Reply:
word.